Changing pycv into a package

Iximiel opened the pull request:
<!--
  Feel free to delete not relevant sections below
-->

##### Description

This should solve #1133, the TL;DR is "calling pycv from the python implementation gives problems"

I rewrote the building part of pyCV to look like a python package.

To install it you need plumed available in your path and you just need to use 
```bash
pip install .
```
in the pycv dir.

To use it, it is slightly less straightforward (these solution **do not** work if you installed with `pip install -e .`):

###### calling plumed from the shell

When you install with pip you get an extra `pycv` executable in the bin of the environment. Calling it will print on the shell the path of the .so that contains the pycv actions.
You can copy-paste that path to your plumed.dat and you are ready to go.

###### calling plumed from python

If you are using plumed in a python script you'll need to have pycv available in the python environment. Then you can get the pycv  object to load with:
```python
from pycv import getPythonCVInterface
from plumed import Plumed
...
plmd = Plumed()
...
plmd.cmd("readInputLine",f"LOAD FILE={getPythonCVInterface()}")
...
```

---

I need to
 - [ ] refine this to compile 100% correctly (aka write a "findPLUMED.cmake")
 - [ ] find if there is a more robust way of passing the path
 - [ ] update the readme
 - [ ] update/add new tests
 - [ ] rewrite its part of the CI (both for the plumed-that-calls-python tests and for the python-calling-plumed-that-calls-python tests)
 - [ ] have some feedback on the user-friendliness
 - [ ] have some feedback on the names

---

[A good number of important projects use scikit-build-core](https://scikit-build-core.readthedocs.io/en/latest/projects.html), among them LAMMPS and Cmake, so it is a project destined to remain.

---

This will unlikely pass the CI for the first few commits

@GiovanniBussi @tonigi, what do you think?

##### Target release

<!-- please tell us where you would like your code to appear (e.g. v2.4): -->
I would like my code to appear in release __2.10__ or __2.11__?

##### Type of contribution

<!--
  Please select the type of your contribution among these:
  (Change [ ] to [X] to tick an option)
-->
- [ ] changes to code or doc authored by PLUMED developers, or additions of code in the core or within the default modules
- [x] changes to a module not authored by you
- [ ] new module contribution or edit of a module authored by you

##### Copyright

<!--
  In case you picked one of the first two choices
  MAKE SURE TO TICK ALSO THE FOLLOWING BOX
-->

- [x] I agree to transfer the copyright of the code I have written to the PLUMED developers or to the author of the code I am modifying.

<!--
  In case you picked the third choice (new module authored by you)
  MAKE SURE TO TICK ALSO THE FOLLOWING BOX
-->

- [ ] the module I added or modified contains a `COPYRIGHT` file with the correct license information. Code should be released under an open source license. I also used the command `cd src && ./header.sh mymodulename` in order to make sure the headers of the module are correct. 

##### Tests

<!--
  Make sure these boxes are checked. For Travis-CI tests, you can wait for them
  to be completed monitoring this page after your pull request has been submitted:
  http://travis-ci.org/plumed/plumed2/pull_requests
-->

- [x] I added a new regtest or modified an existing regtest to validate my changes.
- [ ] I verified that all regtests are passed successfully on [GitHub Actions](https://github.com/plumed/plumed2/actions).

<!--
  After your branch has been merged to the desired branch and then to plumed2/master, and after the
  plumed official manual has been updated, please check out the coverage scan at
  http://www.plumed.org/coverage-master
  In case your new features are not well covered, please try to add more complete regtests.
-->

Iximiel commented:
It works, but I do not understand and I cannot reproduce locally the segmentation faults that are present in the fedora CI.

carlocamilloni commented:
@Iximiel @tonigi @GiovanniBussi is this ready to be merged?

Iximiel commented:
no, @tonigi found some problems on mac, there is the mystery of fedora and the readme is updated to the previous version

tonigi commented:
Even though liked the "standalonecompile" idea,  pip install seems to work better (e.g. in conda). I've pushed updates to the README. For what  regards Mac, I put a note here   https://github.com/tonigi/plumed2/blob/47e9b7128c73b3fa64d03cb66af8a2220c0dbe43/plugins/pycv/README.md?plain=1#L91 , which is hopefully sufficient to make it work.

tonigi commented:
In other words, ok for me to merge (after merging my changes to the readme).

carlocamilloni commented:
so i think that once the conflict and the readme are done i can merge it, correct @Iximiel ?

tonigi commented:
Something that would be nice, IMHO, could be to process the plugins with doxygen and add them to the manual (perhaps under a "Plugin" section).

Iximiel commented:
I just squashed everything and rebased to v2.10.

I need to update the readme with the @tonigi comment and with an installation guide

And check if the CI on fedora had changed idea and now pass

Iximiel commented:
> Something unrelated that would be nice, IMHO, could be to process the plugins with doxygen and add them to the manual (perhaps under a "Plugin" section).

Yep, but that is for another PR

As now I am adding to the manual an example to create the documentation for the package with the built-in python tools (menaning with a discutible old-fashioned graphic style)

Iximiel commented:
the fedora failures *might* be due to the python version being 3.12? Shall I comment the test temporarily under the rug?

Looking at the logs i cann see that Rocky uses an ancient 3.6 (? o_O) 

I forgot to specify that the extra tests, ran with pytest, that I added use the `plumed` python module

carlocamilloni commented:
@Iximiel if it is python 12 the problem than it should be easy to reproduce and it would be better to fix it because isn't that the current supported version? otherwise I am happy to merge it and turn that test off for pycv

Iximiel commented:
I'll try from the wsl if I can reproduce that

tonigi commented:
Im trying to rebuild the fedora dockers to play with. I was hoping it somehow *maybe*  related to python embedded interpreters not honoring their environment paths, hence libpython version mixups.

Iximiel commented:
The crashing tests are to python+plumed+pycv ones, the other, the plumed+pycv look fine

And also on my pc with 3.12.7, like on fedore is segfaulting in the first line here:

```python
def create_plumed_var(plmd: Plumed, name: str, command: str):
    #command="PYCVINTERFACE ATOMS=@mdatoms IMPORT=pycvPerFrameSTR CALCULATE=pydist PREPARE=changeAtom"
    plmd.cmd("readInputLine", name + ": " + command)
    shape = np.zeros(1, dtype=np.int_)
    plmd.cmd("getDataRank " + name, shape)
    data = np.zeros((1))
    plmd.cmd("setMemoryForData " + name, data)
    return data
```
If you are ok I would put this as a known bug and temporarly comment that part of the tests pyenv, then I need to work on this, since looks like it is failing with python 3.10 and so on

tonigi commented:
Ok for me. BTW note that I introduced a problem in committing the "scripts" symlink. It should have been relative, and its creation removed from the dockerfile (now fixed).

tonigi commented:
Locally, docker-building fedora39 and fedora39-pycv is successful...

tonigi commented:
Just an hypothesis: the plumed:fedora39 image pushed on the docker repositories is "bad", and it is not being replaced on our test builds

emilfrost commented:
Hi,

I'm having some trouble installing PYCV this new way. I'm using the v2.10 branch with Python 3.11.3 and GCC 12.3 and have installed PLUMED which seems to work. Doing a simple `pip install .` from the pycv directory throws the error "error: macro names must be identifiers". I attached the full output below. Is it a problem with the compiler version or what do you think?

The old implementation with standalonecompile.sh seems to work fine though from PLUMED release 2.10b, but I want to access the new feature of 'python-calling-plumed-that-calls-python'.

Thanks for the help in advance.

Best regards,
Emil


```
(pycv) [ejfr@slid2 pycv]$ pip install .
Defaulting to user installation because normal site-packages is not writeable
Processing /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy in /home/modules/software/SciPy-bundle/2023.07-gfbf-2023a/lib/python3.11/site-packages (from pycv==0.0.1) (1.25.1)
Requirement already satisfied: pybind11<=2.11.1,>=2.10.3 in /home/modules/software/pybind11/2.11.1-GCCcore-12.3.0/lib/python3.11/site-packages (from pycv==0.0.1) (2.11.1)
Building wheels for collected packages: pycv
  Building wheel for pycv (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pycv (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [59 lines of output]
      *** scikit-build-core 0.11.0 using CMake 3.26.5 (wheel)
      *** Configuring CMake...
      loading initial cache file /tmp/tmpfqx1hi66/build/CMakeInit.txt
      -- The CXX compiler identification is GNU 12.3.0
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /home/modules/software/GCCcore/12.3.0/bin/g++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Everithing should work fine if you are in the same environment in which you have compiled plumed
      -- Found Python: /home/modules/software/Python/3.11.3-GCCcore-12.3.0/bin/python (found version "3.11.3") found components: Interpreter Development Development.Module Development.Embed
      -- Performing Test HAS_FLTO
      -- Performing Test HAS_FLTO - Success
      -- Found pybind11: /tmp/pip-build-env-sexz_gbc/overlay/lib/python3.11/site-packages/pybind11/include (found version "2.11.1")
      -- Found PkgConfig: /usr/bin/pkg-config (found version "1.4.2")
      -- Checking for module 'plumedInternals'
      --   Found plumedInternals, version 2.10b
      -- plumed not found via pkgconfig, trying executable
      -- Configuring plumed from executable
      -- Found MPI_CXX: /home/modules/software/OpenMPI/4.1.5-GCC-12.3.0/lib/libmpi.so (found version "3.1")
      -- Found MPI: TRUE (found version "3.1")
      -- Performing Test USE_NO_GNU_UNIQUE
      -- Performing Test USE_NO_GNU_UNIQUE - Success
      -- Configuring done (2.2s)
      -- Generating done (0.0s)
      -- Build files have been written to: /tmp/tmpfqx1hi66/build
      *** Building project with Ninja...
      [1/6] Building CXX object CMakeFiles/PythonCVInterface.dir/src/ActionWithPython.cpp.o
      FAILED: CMakeFiles/PythonCVInterface.dir/src/ActionWithPython.cpp.o
      /home/modules/software/GCCcore/12.3.0/bin/g++ -D-I/home/modules/energy/software/Boost/1.82.0-GCC-12.3.0/include -I/home/modules/software/Python/3.11.3-GCCcore-12.3.0/include -I/home/modules/energy/software/GSL/2.7-GCC-12.3.0/include -I/home/modules/software/zlib/1.2.13-GCCcore-12.3.0/include -I/home/modules/software/FFTW.MPI/3.3.10-gompi-2023a/include -I/home/modules/software/FlexiBLAS/3.3.1-GCC-12.3.0/include -I/home/modules/software/FlexiBLAS/3.3.1-GCC-12.3.0/include/flexiblas  -DHAVE_INTTYPES_H="1 " -DHAVE_MEMORY_H="1 " -DHAVE_STDINT_H="1 " -DHAVE_STDLIB_H="1 " -DHAVE_STRINGS_H="1 " -DHAVE_STRING_H="1 " -DHAVE_SYS_STAT_H="1 " -DHAVE_SYS_TYPES_H="1 " -DHAVE_UNISTD_H="1 " -DNDEBUG="1 " -DPACKAGE_BUGREPORT="\"\" " -DPACKAGE_NAME="\"PLUMED\" " -DPACKAGE_STRING="\"PLUMED 2\" " -DPACKAGE_TARNAME="\"plumed\" " -DPACKAGE_URL="\"\" " -DPACKAGE_VERSION="\"2\" " -DPythonCVInterface_EXPORTS -DSTDC_HEADERS="1 " -D_REENTRANT="1 " -D__PLUMED_CXX17_FILESYSTEM="1 " -D__PLUMED_HAS_ASMJIT="1 " -D__PLUMED_HAS_BOOST_SERIALIZATION="1 " -D__PLUMED_HAS_DLADDR="1 " -D__PLUMED_HAS_DLOPEN="1 " -D__PLUMED_HAS_EXECINFO="1 " -D__PLUMED_HAS_EXTERNAL_BLAS="1 " -D__PLUMED_HAS_EXTERNAL_LAPACK="1 " -D__PLUMED_HAS_FFTW="1 " -D__PLUMED_HAS_GSL="1 " -D__PLUMED_HAS_MOLFILE_PLUGINS="1 " -D__PLUMED_HAS_MPI="1 " -D__PLUMED_HAS_PYTHON="1 " -D__PLUMED_HAS_RTLD_DEFAULT="1 " -D__PLUMED_HAS_SUBPROCESS="1 " -D__PLUMED_HAS_ZLIB="1 " -D__PLUMED_PROGRAM_NAME="\"plumed\" " -D__PLUMED_SOEXT=\"so\" -D__PLUMED_WRAPPER_LIBCXX11="1 " -I/home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src -I/home/modules/energy/software/PLUMED/2.10-foss-2023a-669a52ce3324fe81b079935a870360c68ce507f4/include -isystem /tmp/pip-build-env-sexz_gbc/overlay/lib/python3.11/site-packages/pybind11/include -isystem /home/modules/software/Python/3.11.3-GCCcore-12.3.0/include/python3.11 -isystem /home/modules/software/OpenMPI/4.1.5-GCC-12.3.0/include -fno-gnu-unique -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/PythonCVInterface.dir/src/ActionWithPython.cpp.o -MF CMakeFiles/PythonCVInterface.dir/src/ActionWithPython.cpp.o.d -o CMakeFiles/PythonCVInterface.dir/src/ActionWithPython.cpp.o -c /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/ActionWithPython.cpp
      <command-line>: error: macro names must be identifiers
      [2/6] Building CXX object CMakeFiles/PythonCVInterface.dir/src/PythonFunction.cpp.o
      FAILED: CMakeFiles/PythonCVInterface.dir/src/PythonFunction.cpp.o
      /home/modules/software/GCCcore/12.3.0/bin/g++ -D-I/home/modules/energy/software/Boost/1.82.0-GCC-12.3.0/include -I/home/modules/software/Python/3.11.3-GCCcore-12.3.0/include -I/home/modules/energy/software/GSL/2.7-GCC-12.3.0/include -I/home/modules/software/zlib/1.2.13-GCCcore-12.3.0/include -I/home/modules/software/FFTW.MPI/3.3.10-gompi-2023a/include -I/home/modules/software/FlexiBLAS/3.3.1-GCC-12.3.0/include -I/home/modules/software/FlexiBLAS/3.3.1-GCC-12.3.0/include/flexiblas  -DHAVE_INTTYPES_H="1 " -DHAVE_MEMORY_H="1 " -DHAVE_STDINT_H="1 " -DHAVE_STDLIB_H="1 " -DHAVE_STRINGS_H="1 " -DHAVE_STRING_H="1 " -DHAVE_SYS_STAT_H="1 " -DHAVE_SYS_TYPES_H="1 " -DHAVE_UNISTD_H="1 " -DNDEBUG="1 " -DPACKAGE_BUGREPORT="\"\" " -DPACKAGE_NAME="\"PLUMED\" " -DPACKAGE_STRING="\"PLUMED 2\" " -DPACKAGE_TARNAME="\"plumed\" " -DPACKAGE_URL="\"\" " -DPACKAGE_VERSION="\"2\" " -DPythonCVInterface_EXPORTS -DSTDC_HEADERS="1 " -D_REENTRANT="1 " -D__PLUMED_CXX17_FILESYSTEM="1 " -D__PLUMED_HAS_ASMJIT="1 " -D__PLUMED_HAS_BOOST_SERIALIZATION="1 " -D__PLUMED_HAS_DLADDR="1 " -D__PLUMED_HAS_DLOPEN="1 " -D__PLUMED_HAS_EXECINFO="1 " -D__PLUMED_HAS_EXTERNAL_BLAS="1 " -D__PLUMED_HAS_EXTERNAL_LAPACK="1 " -D__PLUMED_HAS_FFTW="1 " -D__PLUMED_HAS_GSL="1 " -D__PLUMED_HAS_MOLFILE_PLUGINS="1 " -D__PLUMED_HAS_MPI="1 " -D__PLUMED_HAS_PYTHON="1 " -D__PLUMED_HAS_RTLD_DEFAULT="1 " -D__PLUMED_HAS_SUBPROCESS="1 " -D__PLUMED_HAS_ZLIB="1 " -D__PLUMED_PROGRAM_NAME="\"plumed\" " -D__PLUMED_SOEXT=\"so\" -D__PLUMED_WRAPPER_LIBCXX11="1 " -I/home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src -I/home/modules/energy/software/PLUMED/2.10-foss-2023a-669a52ce3324fe81b079935a870360c68ce507f4/include -isystem /tmp/pip-build-env-sexz_gbc/overlay/lib/python3.11/site-packages/pybind11/include -isystem /home/modules/software/Python/3.11.3-GCCcore-12.3.0/include/python3.11 -isystem /home/modules/software/OpenMPI/4.1.5-GCC-12.3.0/include -fno-gnu-unique -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/PythonCVInterface.dir/src/PythonFunction.cpp.o -MF CMakeFiles/PythonCVInterface.dir/src/PythonFunction.cpp.o.d -o CMakeFiles/PythonCVInterface.dir/src/PythonFunction.cpp.o -c /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonFunction.cpp
      <command-line>: error: macro names must be identifiers
      In file included from /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonFunction.cpp:17:
      /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonFunction.h:27:7: warning: ‘PLMD::pycv::PythonFunction’ declared with greater visibility than the type of its field ‘PLMD::pycv::PythonFunction::pyModule’ [-Wattributes]
         27 | class PythonFunction :
            |       ^~~~~~~~~~~~~~
      /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonFunction.h:27:7: warning: ‘PLMD::pycv::PythonFunction’ declared with greater visibility than the type of its field ‘PLMD::pycv::PythonFunction::pyCalculate’ [-Wattributes]
      [3/6] Building CXX object CMakeFiles/PythonCVInterface.dir/src/PythonCVInterface.cpp.o
      FAILED: CMakeFiles/PythonCVInterface.dir/src/PythonCVInterface.cpp.o
      /home/modules/software/GCCcore/12.3.0/bin/g++ -D-I/home/modules/energy/software/Boost/1.82.0-GCC-12.3.0/include -I/home/modules/software/Python/3.11.3-GCCcore-12.3.0/include -I/home/modules/energy/software/GSL/2.7-GCC-12.3.0/include -I/home/modules/software/zlib/1.2.13-GCCcore-12.3.0/include -I/home/modules/software/FFTW.MPI/3.3.10-gompi-2023a/include -I/home/modules/software/FlexiBLAS/3.3.1-GCC-12.3.0/include -I/home/modules/software/FlexiBLAS/3.3.1-GCC-12.3.0/include/flexiblas  -DHAVE_INTTYPES_H="1 " -DHAVE_MEMORY_H="1 " -DHAVE_STDINT_H="1 " -DHAVE_STDLIB_H="1 " -DHAVE_STRINGS_H="1 " -DHAVE_STRING_H="1 " -DHAVE_SYS_STAT_H="1 " -DHAVE_SYS_TYPES_H="1 " -DHAVE_UNISTD_H="1 " -DNDEBUG="1 " -DPACKAGE_BUGREPORT="\"\" " -DPACKAGE_NAME="\"PLUMED\" " -DPACKAGE_STRING="\"PLUMED 2\" " -DPACKAGE_TARNAME="\"plumed\" " -DPACKAGE_URL="\"\" " -DPACKAGE_VERSION="\"2\" " -DPythonCVInterface_EXPORTS -DSTDC_HEADERS="1 " -D_REENTRANT="1 " -D__PLUMED_CXX17_FILESYSTEM="1 " -D__PLUMED_HAS_ASMJIT="1 " -D__PLUMED_HAS_BOOST_SERIALIZATION="1 " -D__PLUMED_HAS_DLADDR="1 " -D__PLUMED_HAS_DLOPEN="1 " -D__PLUMED_HAS_EXECINFO="1 " -D__PLUMED_HAS_EXTERNAL_BLAS="1 " -D__PLUMED_HAS_EXTERNAL_LAPACK="1 " -D__PLUMED_HAS_FFTW="1 " -D__PLUMED_HAS_GSL="1 " -D__PLUMED_HAS_MOLFILE_PLUGINS="1 " -D__PLUMED_HAS_MPI="1 " -D__PLUMED_HAS_PYTHON="1 " -D__PLUMED_HAS_RTLD_DEFAULT="1 " -D__PLUMED_HAS_SUBPROCESS="1 " -D__PLUMED_HAS_ZLIB="1 " -D__PLUMED_PROGRAM_NAME="\"plumed\" " -D__PLUMED_SOEXT=\"so\" -D__PLUMED_WRAPPER_LIBCXX11="1 " -I/home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src -I/home/modules/energy/software/PLUMED/2.10-foss-2023a-669a52ce3324fe81b079935a870360c68ce507f4/include -isystem /tmp/pip-build-env-sexz_gbc/overlay/lib/python3.11/site-packages/pybind11/include -isystem /home/modules/software/Python/3.11.3-GCCcore-12.3.0/include/python3.11 -isystem /home/modules/software/OpenMPI/4.1.5-GCC-12.3.0/include -fno-gnu-unique -O3 -DNDEBUG -fPIC -MD -MT CMakeFiles/PythonCVInterface.dir/src/PythonCVInterface.cpp.o -MF CMakeFiles/PythonCVInterface.dir/src/PythonCVInterface.cpp.o.d -o CMakeFiles/PythonCVInterface.dir/src/PythonCVInterface.cpp.o -c /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonCVInterface.cpp
      <command-line>: error: macro names must be identifiers
      In file included from /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonCVInterface.cpp:17:
      /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonCVInterface.h:32:7: warning: ‘PLMD::pycv::PythonCVInterface’ declared with greater visibility than the type of its field ‘PLMD::pycv::PythonCVInterface::pyModule’ [-Wattributes]
         32 | class PythonCVInterface : public Colvar, public ActionWithPython {
            |       ^~~~~~~~~~~~~~~~~
      /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonCVInterface.h:32:7: warning: ‘PLMD::pycv::PythonCVInterface’ declared with greater visibility than the type of its field ‘PLMD::pycv::PythonCVInterface::pyCalculate’ [-Wattributes]
      /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonCVInterface.h:32:7: warning: ‘PLMD::pycv::PythonCVInterface’ declared with greater visibility than the type of its field ‘PLMD::pycv::PythonCVInterface::pyPrepare’ [-Wattributes]
      /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonCVInterface.h:32:7: warning: ‘PLMD::pycv::PythonCVInterface’ declared with greater visibility than the type of its field ‘PLMD::pycv::PythonCVInterface::pyUpdate’ [-Wattributes]
      /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PythonCVInterface.h:32:7: warning: ‘PLMD::pycv::PythonCVInterface’ declared with greater visibility than the type of its field ‘PLMD::pycv::PythonCVInterface::dataContainer’ [-Wattributes]
      [4/6] Building CXX object CMakeFiles/plumedCommunications.dir/src/PlumedPythonEmbeddedModule.cpp.o
      FAILED: CMakeFiles/plumedCommunications.dir/src/PlumedPythonEmbeddedModule.cpp.o
      /home/modules/software/GCCcore/12.3.0/bin/g++ -D-I/home/modules/energy/software/Boost/1.82.0-GCC-12.3.0/include -I/home/modules/software/Python/3.11.3-GCCcore-12.3.0/include -I/home/modules/energy/software/GSL/2.7-GCC-12.3.0/include -I/home/modules/software/zlib/1.2.13-GCCcore-12.3.0/include -I/home/modules/software/FFTW.MPI/3.3.10-gompi-2023a/include -I/home/modules/software/FlexiBLAS/3.3.1-GCC-12.3.0/include -I/home/modules/software/FlexiBLAS/3.3.1-GCC-12.3.0/include/flexiblas  -DHAVE_INTTYPES_H="1 " -DHAVE_MEMORY_H="1 " -DHAVE_STDINT_H="1 " -DHAVE_STDLIB_H="1 " -DHAVE_STRINGS_H="1 " -DHAVE_STRING_H="1 " -DHAVE_SYS_STAT_H="1 " -DHAVE_SYS_TYPES_H="1 " -DHAVE_UNISTD_H="1 " -DNDEBUG="1 " -DPACKAGE_BUGREPORT="\"\" " -DPACKAGE_NAME="\"PLUMED\" " -DPACKAGE_STRING="\"PLUMED 2\" " -DPACKAGE_TARNAME="\"plumed\" " -DPACKAGE_URL="\"\" " -DPACKAGE_VERSION="\"2\" " -DSTDC_HEADERS="1 " -D_REENTRANT="1 " -D__PLUMED_CXX17_FILESYSTEM="1 " -D__PLUMED_HAS_ASMJIT="1 " -D__PLUMED_HAS_BOOST_SERIALIZATION="1 " -D__PLUMED_HAS_DLADDR="1 " -D__PLUMED_HAS_DLOPEN="1 " -D__PLUMED_HAS_EXECINFO="1 " -D__PLUMED_HAS_EXTERNAL_BLAS="1 " -D__PLUMED_HAS_EXTERNAL_LAPACK="1 " -D__PLUMED_HAS_FFTW="1 " -D__PLUMED_HAS_GSL="1 " -D__PLUMED_HAS_MOLFILE_PLUGINS="1 " -D__PLUMED_HAS_MPI="1 " -D__PLUMED_HAS_PYTHON="1 " -D__PLUMED_HAS_RTLD_DEFAULT="1 " -D__PLUMED_HAS_SUBPROCESS="1 " -D__PLUMED_HAS_ZLIB="1 " -D__PLUMED_PROGRAM_NAME="\"plumed\" " -D__PLUMED_SOEXT=\"so\" -D__PLUMED_WRAPPER_LIBCXX11="1 " -DplumedCommunications_EXPORTS -I/home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src -I/home/modules/energy/software/PLUMED/2.10-foss-2023a-669a52ce3324fe81b079935a870360c68ce507f4/include -isystem /home/modules/software/Python/3.11.3-GCCcore-12.3.0/include/python3.11 -isystem /tmp/pip-build-env-sexz_gbc/overlay/lib/python3.11/site-packages/pybind11/include -isystem /home/modules/software/OpenMPI/4.1.5-GCC-12.3.0/include -fno-gnu-unique -O3 -DNDEBUG -fPIC -fvisibility=hidden -flto -fno-fat-lto-objects -MD -MT CMakeFiles/plumedCommunications.dir/src/PlumedPythonEmbeddedModule.cpp.o -MF CMakeFiles/plumedCommunications.dir/src/PlumedPythonEmbeddedModule.cpp.o.d -o CMakeFiles/plumedCommunications.dir/src/PlumedPythonEmbeddedModule.cpp.o -c /home/energy/ejfr/git_packages/test/plumed2/plugins/pycv/src/PlumedPythonEmbeddedModule.cpp
      <command-line>: error: macro names must be identifiers
      ninja: build stopped: subcommand failed.
      
      *** CMake build failed
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pycv
Failed to build pycv
ERROR: Failed to build installable wheels for some pyproject.toml based projects (pycv)
```
